### PR TITLE
Revert [now-cli][now-routing-utils] Add support for v4 builders #3311

### PR DESCRIPTION
**Pull request by styfle**
Reverts #3311

Previously, we thought we were going to use version 4 to support builder rewrites/redirects => routes. But that is no longer necessary. So instead version 4 will be builders that accept multiple file entrypoints.

Until v4 can be implemented, this reverts v4 changes.

Additionally, I added a helpful error message when the CLI does not support the builder version. In a future PR, we will add better version locking to avoid this type of version mismatch.

### CODE REVIEW

File: packages/now-cli/src/util/dev/builder.ts [hunk 1]
```diff
@@ -268,27 +265,16 @@ export async function executeBuild(
...
-      routes,
       output: {
         [entrypoint]: output,
       },
     } as BuildResult;
   } else {
-    result = buildResultOrOutputs as BuildResult;
+    throw new Error(
+      `Now CLI does not support builder version: ${builder.version}`
+    );
```
Comment (styfle) [inline packages/now-cli/src/util/dev/builder.ts:275]: This is the one new part that will provide a helpful error when we do bump the version.

**Review by AndyBitz — APPROVED** [review]

File: packages/now-cli/src/util/dev/builder.ts [hunk 2]
```diff
@@ -522,3 +508,11 @@ export async function shutdownBuilder(
 
   await Promise.all(ops);
 }
+
+function findMatchingLambda(src: string, output: BuildResultV4) {
```
Comment (lucleray) [inline packages/now-cli/src/util/dev/builder.ts]: Where is this function used?

File: packages/now-cli/src/util/dev/builder.ts [hunk 3]
```diff
@@ -31,9 +31,8 @@ import {
   BuildResultV3,
   BuildResultV4,
```
Comment (lucleray) [inline packages/now-cli/src/util/dev/builder.ts]: Do we still need to import this?

File: packages/now-cli/src/util/dev/builder.ts (hunk 2 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/builder.ts]: Forgot to revert that, done ✅

File: packages/now-cli/src/util/dev/builder.ts (hunk 3 above)
Comment (styfle) [reply packages/now-cli/src/util/dev/builder.ts]: Removed it, thanks

**Review by lucleray — APPROVED** [review]

**Pull request merged**